### DISCUSSION TITLE
Reduce minimum current to 5A

### DISCUSSION THREAD

**Discussion by negronoir** [dingo35/SmartEVSE-3.5#30] [Ideas]
Hello,

Another idea is to lower min current to 5A. I know that this is out of standard 6A, but some brands like Tesla, allow to charge at 5A.

If fact, changing few lines, it works. I rebuilt Serkri v1.7.3 & v1.8.0 to allow 5A and I've been using those custom firms for some months WO any issue related to 5A

Thanks

**Comment by dingo35** [comment]
Yes I know it works with Tesla. But all standards on EV charging demand minimum charge current is 6A, and we would hate to have to support users who lower the current to 5A and then get problems with their EV's.

So if you want to deviate from the standards, you can change the code to your likings, that is the great thing of open source!

**Reply by negronoir** [reply]
Haha you are very hard to convince to reduce to 5A :-)

Okay, I'll continue building my own versions. Thanks anyway!

But as I test each new released version and also the related code is a bit messy (it is not enough to change the MIN_CURRENT constant because the value "6" is hardcoded in a couple of places, It's a little effort that I'd like to avoid.

So please, in next release change this source code lines

evse.h
Line 493: `{"MIN",     "MIN Charge Current the EV will accept (per phase)",  6, 16, MIN_CURRENT},`
change to: `{"MIN",     "MIN Charge Current the EV will accept (per phase)",  MIN_CURRENT, 16, MIN_CURRENT},`

evse.cpp
Line 471: `if ((current >= 60) && (current <= 510)) DutyCycle = current / 0.6;`
change to: `if ((current >= (MIN_CURRENT * 10)) && (current <= 510)) DutyCycle = current / 0.6;`

Line 3953: `if(current >= 6 && current <= 16 && LoadBl < 2) {`
change to: `if(current >= MIN_CURRENT && current <= 16 && LoadBl < 2) {`

So, in this way I can rebuild only changing constant MIN_CURRENT

Thanks!

**Comment by dingo35** [comment]
That is a great point, happy to help with that!

Now as you might or might not now, we are now fully concentrating on bringing out version 3.5.0 , so I like to put this in version 3.5.1 .

How about if you wait (a few weeks) until our 3.5.0 is stabilized and then you bring out a Pull Request? I would hate doing it now because you would keep updating your pull request to changes in our repo...

Ok with that?




"negronoir" ***@***.***> schreef op 7 maart 2024 om 09:07:

**Reply by negronoir** [reply]
Of course. Perfect!

I'll remember you again after v3.5.0

Many thanks for your support!

**Comment by negronoir** [comment]
Hi!

I barely use GitHub so I'm not sure about creating pull requests properly.

To tidy up "min current" related source code for allowing to build 5A custom versions please, in next release change this source code lines from RC2 I've updated lines):

evse.h
Line 494: `{"MIN",     "MIN Charge Current the EV will accept (per phase)",  6, 16, MIN_CURRENT},`
change to: `{"MIN",     "MIN Charge Current the EV will accept (per phase)",  MIN_CURRENT, 16, MIN_CURRENT},`

evse.cpp
Line 472: `if ((current >= 60) && (current <= 510)) DutyCycle = current / 0.6;`
change to: `if ((current >= (MIN_CURRENT * 10)) && (current <= 510)) DutyCycle = current / 0.6;`

Line 3974: `if(current >= 6 && current <= 16 && LoadBl < 2) {`
change to: `if(current >= MIN_CURRENT && current <= 16 && LoadBl < 2) {`

In this way we can rebuild only changing constant MIN_CURRENT

Thanks!

**Comment by dingo35** [comment]
You're a bit early, but I included the patch in my personal master, so it will included next release; I even added some code so now if you compile it like this:
PLATFORMIO_BUILD_FLAGS='-DMIN_CURRENT=5' pio run 

...you don't even have to modify code.

You're welcome!

**Reply by negronoir** [reply]
Wow!

Many thanks for your work!

**Comment by adjego** [comment]
Is there a build guide available for this version with minimal 5A? I'm also interested in this lower min_current :)

**Reply by negronoir** [reply]
Yes. It's quite easy.

You only need to modify platformio.ini file and add a build flag -DMIN_CURRENT=5 into [env:release] section:

This is the example from my current v3.6.3 platformio.ini file

```
[env:release]
lib_deps =
    https://github.com/tzapu/WiFiManager.git
    miq19/eModbus@1.7.1
    bblanchon/ArduinoJson@^6.21.4
build_flags =
    -DCORE_DEBUG_LEVEL=0
    -DLOG_LEVEL=0
    -DMG_ENABLE_PACKED_FS=1
    -DMG_TLS=MG_TLS_MBED
    -Wall
    -Wextra
    -Wunused-variable
    -DNDEBUG
    -DMG_ENABLE_LOG=0
    -DMIN_CURRENT=5
```

Then, you can build the project

**Reply by adjego** [reply]
En with which tool? Windows program? 
Edit: PlatformIO is see now :)

**Comment by Ferforfaen** [comment]
Hi Dingo

Thank you for the great job. I tried adding the -DMIN_CURRENT=5 in my platfomio.ini and build the firmware. I can now lower the setting to 5A. But it will not go below 6A, when controlled by the EVSE in solar. When i reduce max to 5A in the car, it goes down to 5A. 

Regards Tommy

**Comment by dingo35** [comment]
Does it go down to 5A when its controlled by SmartEVSE in Smart Mode?

**Reply by Ferforfaen** [reply]
I tried to reduce my mains setting to get it down, but it shut off when it was forced below 6A. So no.

**Comment by dingo35** [comment]
So then your EV is the problem, not the EVSE?
Chsrging below 6A is outside of the protocol specifications, many EVs cannot handle the pulse on the CP line lower then 6A. If the EV itself decides to lower its charge current the pulse width on the CP line stays within specification.

So dont fiddle with MinCurrent if your EV cant handle it and if you are not an expert.

**Reply by Ferforfaen** [reply]
Well... maybe... It can reduce the current to 5a it self? No idea if the CP line pulse is correct. Could put a scope on, and see if there is a difference between 5A and 6A. I might suspect smartevse not lowering below 6A
